### PR TITLE
docs: Add patch level release process

### PR DESCRIPTION
**Pull request by fabiand**
Stable branches seem to be used by now.
This is some documentation to formalize the patch level
release process which is happening if new releases appear
on a stable branch.

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by fabiand** [comment]
This is the process I used for all patch level releases so far.

**Comment by fabiand** [comment]
… thus I'd recommend to merge it this way (as it reflects how it was done), and do any changes according to how the process shall be changed.

File: docs/release.md [hunk 1]
```diff
@@ -119,3 +119,23 @@ git push $YOUR_REMOTE release-0.6:release-0.6-aParticularFix
...
 against the correct target branch in GitHub (in this case the target branch
 is `release-0.6`).
+
+Releasing patch versions
+------------------------
+
+Releases on the stable branch only increment the patch level.
+The release itself is only a evsigned tag as it's used for minor releases as well.
+
+1. Tag the commit using `git evtag sign $TAG` (which is a signed and annotated
```
Comment (stu-gott) [inline docs/release.md]: Successful execution of this command appears to rely on two things.

* git-evtag is installed
* gpg is set up with at least one keypair that can be used to do the signing

Assuming that's true, will we want to have some formalized process of distributing public keys?

Comment (fabiand) [reply docs/release.md]: Good idea, I'll link to
https://help.github.com/articles/adding-a-new-gpg-key-to-your-github-account/

Comment (fabiand) [reply docs/release.md]: Done.

**Pull request merged**